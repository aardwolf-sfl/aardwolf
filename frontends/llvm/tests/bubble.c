void bubble_sort(int *values, unsigned n)
{
    unsigned i;
    int temp;
    unsigned char swapped = 0;

    do {
        for (i = 0; i < n - 1; i++) {
            swapped = 0;

            if (values[i] > values[i + 1]) {
                temp = values[i];
                values[i] = values[i + 1];
                values[i + 1] = temp;
                swapped = 1;
            }
        }

        n = n - 1;
    } while (swapped);
}
