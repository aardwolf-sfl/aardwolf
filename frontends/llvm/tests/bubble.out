
function: bubble_sort

#1 -> #2  ::  defs: %1 / uses:  [@1 1:23-1:23]  { arg }
#2 -> #3  ::  defs: %2 / uses:  [@1 1:40-1:40]  { arg }
#3 -> #4  ::  defs: %3 / uses:  [@1 5:19-5:19]
#4 -> #5  ::  defs: %4 / uses:  [@1 8:16-8:16]
#5 -> #13, #6  ::  defs:  / uses: %2, %4 [@1 8:9-8:9]
#6 -> #7  ::  defs: %3 / uses:  [@1 9:21-9:21]
#7 -> #12, #8  ::  defs:  / uses: %1[%4] [@1 11:17-11:17]
#8 -> #9  ::  defs: %5 / uses: %1[%4] [@1 12:22-12:22]
#9 -> #10  ::  defs: %1[%4] / uses: %1[%4] [@1 13:27-13:27]
#10 -> #11  ::  defs: %1[%4] / uses: %5 [@1 14:31-14:31]
#11 -> #12  ::  defs: %3 / uses:  [@1 15:25-15:25]
#12 -> #5  ::  defs: %4 / uses: %4 [@1 8:33-8:33]
#13 -> #14  ::  defs: %2 / uses: %2 [@1 19:11-19:11]
#14 -> #15, #4  ::  defs:  / uses: %3 [@1 20:5-20:5]
#15 ->   ::  defs:  / uses:  [@1 21:1-21:1]  { ret }

@1 = tests/bubble.c
